client/run.go

```diff
@@ -290,6 +303,67 @@ func (cli *HyperClient) JsonFromCmdline(cmdArgs, cmdEnvs, cmdPortmaps []string,
 	return string(jsonString), nil
 }
 
+func parseVolume(volStr string) (*pod.UserVolume, *pod.UserVolumeReference, error) {
+
+	var (
+		srcName   string
+		destPath  string
+		volName   string
+		readOnly  = false
+		volDriver = "vfs"
+	)
+
+	fields := strings.Split(volStr, ":")
+	if len(fields) == 3 {
+		// cmd: -v host-src:container-dest:rw
+		srcName = fields[0]
+		destPath = fields[1]
+		if fields[2] != "ro" && fields[2] != "rw" {
+			return nil, nil, fmt.Errorf("flag only support(ro or rw): --volume")
+		}
+		if fields[2] == "ro" {
+			readOnly = true
+		}
+	} else if len(fields) == 2 {
+		// cmd: -v host-src:container-dest
+		srcName = fields[0]
+		destPath = fields[1]
+	} else if len(fields) == 1 {
+		// -v container-dest
+		destPath = fields[0]
+	} else {
+		return nil, nil, fmt.Errorf("flag format should be like : --volume=[host-src:]container-dest[:rw|ro]")
+	}
+
+	_, volName = path.Split(srcName)
```

Comment:
here are two "_, volName =", the value depends on the "if else" branches, could you move it into the branch?